lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -1463,6 +1463,17 @@ public void setHint(@Nullable CharSequence hint) {
     }
   }
 
+  /**
+   * Set the hint to be displayed in the floating label, if enabled, using the given resource id.
+   *
+   * @see #setHintEnabled(boolean)
+   * @see #setHint(CharSequence)
+   * @param textHintId The resource id of the text to display in the floating label
+   */
+  public void setHint(@StringRes int textHintId) {
+    setHint(getContext().getText(textHintId));
```

Comment:
Looking how we do it in similar methods, please call setHint(textHintId != 0 ? getResources().getText(textHintId) : null)

Reply:
Done